tutorials/persistance-comparison-series/BinaryWriterReader/binary_reader_writer.md

```diff
@@ -0,0 +1,426 @@
+
+The more complex data gets, the more complicated it will be to make sure there are no mistakes in the structure when reading or writing it. Different types, adding and removing variables, changing the structure. The more data we want to add to this file, the more it makes sense to think about alternatatives. For this tutorial, we will stick with the `BinaryReader` and `BinaryWriter` and see what we can do to decrease the complexity a bit when adding more data.
+
+One of those options will be shown in the next section.
+
+## More complex data
+
+(See `BinaryReaderWriterExampleJson.cs` in the repository for the finished version.)
+
+JSON is a very common approach when saving structured data. It's easy to use and there are frameworks for almost every language. The .NET framework provides a [`JsonSerializer`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer?view=net-6.0). Unity has its own version of it: [`JsonUtility`](https://docs.unity3d.com/ScriptReference/JsonUtility.html).
+
+As you can see in the documentation, the functionality boils down to these three methods:
+
+- *FromJson()*: Create an object from its JSON representation.
+- *FromJsonOverwrite()*: Overwrite data in an object by reading from its JSON representation.
+- *ToJson()*: Generate a JSON representation of the public fields of an object.
+
+The `JsonUtility` transforms JSON into objects and back. Therefore, our first change to the previous section is to define such an object with public fields:
+
+```cs
+private class HitCount
+{
+    public int Unmodified;
+    public int Shift;
+    public int Control;
+}
+```
+
+The class itself can be `private` and just be added inside the `BinaryReaderWriterExampleJson` class, but its fields need to be public.
+
+As before, we use a different file to save this data. Update the filename to:
+
+```cs
+private const string HitCountFile = "BinaryReaderWriterExampleJson";
+```
+
+When saving the data, we will use the same `Update()` method as before to detect which key was pressed.
+
+The first part of `OnMouseDown()` (1) can stay the same as well, since this part only increments the hit count depending on the modifier used.
+
+```cs
+private void OnMouseDown() // 1
+{
+    // Check if a key was pressed.
+    switch (modifier)
+    {
+        case KeyCode.LeftShift:
+            // Increment the Shift hit count.
+            hitCountShift++;
+            break;
+        case KeyCode.LeftControl:
+            // Increment the Control hit count.
+            hitCountControl++;
+            break;
+        default:
+            // If neither Shift nor Control was held, we increment the unmodified hit count.
+            hitCountUnmodified++;
+            break;
+    }
+
+    // 2
+    // Create a new HitCount object to hold this data.
+    var updatedCount = new HitCount
```

Comment:
Since you've been using the **target-typed new**, shouldn't you use the same style here?

Reply:
Agree! Updated in ac15b8e9decad9f95666d5e37b811f651dc4a7dc